style: Update css-align to the spec (mostly)

emilio opened the pull request:
This is on top of #19850.

This fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1430817, and updates us
to the current version of the css-align spec.

The only remaining change is the justify-items: auto FIXME.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/19851)
<!-- Reviewable:end -->

emilio committed:
style: Make content distribution parsing know the axis it's parsed for.

MozReview-Commit-ID: LMPXVtKU1mq

emilio committed:
style: Restrict <baseline-position> and <content-position> depending on the axis in content distribution properties.

This fixes https://bugzilla.mozilla.org/show_bug.cgi?id=1430817, and updates us
to the current version of the css-align spec.

MozReview-Commit-ID: LtBcdqYJeK

highfive commented:
Heads up! This PR modifies the following files:
 * @bholley: components/style/values/computed/align.rs, components/style/gecko/generated/structs.rs, components/style/values/specified/align.rs, components/style/values/specified/mod.rs, components/style/values/computed/mod.rs and 2 more
 * @canaltinova: components/style/values/computed/align.rs, components/style/gecko/generated/structs.rs, components/style/values/specified/align.rs, components/style/values/specified/mod.rs, components/style/values/computed/mod.rs and 2 more

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify style code, but no tests are modified. Please consider adding a test!

emilio commented:
r? @MatsPalmgren and @nox 

 * Running through try now to see how many tests do I need to update.

emilio committed:
style: Fix the position of the <overflow-position> in content distribution shorthands, and remove fallback.

MozReview-Commit-ID: 4rPICzZ5gMn

emilio committed:
style: Update align-self / justify-self to the spec too.

MozReview-Commit-ID: 8JKF5ucCbYm

emilio committed:
style: Serialize <overflow-position> before other align bits.

Otherwise the serialisation wouldn't roundtrip with the new syntax, which fixes
the position of <overflow-position>.

Also make Servo and Gecko agree on whether to serialize "unsafe".

Bug: 1430817
MozReview-Commit-ID: L3GSMk5pZ3F

bors-servo commented:
:umbrella: The latest upstream changes (presumably #19850) made this pull request unmergeable. Please resolve the merge conflicts.

emilio commented:
r? @nox

emilio committed:
style: Properly compare the value of the alignment flags.

What this did before happens to be ok for these constants, but it's not great,
because it depends on their particular value.

nox commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit b7238c4 has been approved by `nox`

<!-- @bors-servo r=nox b7238c43cd6899cb1530f8ece8d6d3f3c17e154d -->

bors-servo commented:
:hourglass: Testing commit b7238c43cd6899cb1530f8ece8d6d3f3c17e154d with merge 33818f6b404a5f2fc446da68978e9f3ce9bf2f7a...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/10556), [arm32](http://build.servo.org/builders/arm32/builds/10799), [arm64](http://build.servo.org/builders/arm64/builds/10828), [linux-dev](http://build.servo.org/builders/linux-dev/builds/10850), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/7716), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/7732), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/10925), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/3323), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/3347), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/7589), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/7660), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/3420), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/3367), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/6231)
Approved by: nox
Pushing 33818f6b404a5f2fc446da68978e9f3ce9bf2f7a to master...
